server/lib/reporting/extended-information.ts

```diff
@@ -0,0 +1,895 @@
+import summaryTablesDefinitions from './summary-tables-definitions';
+
+
+
+/**
+ * This build the agents table
+ * @param {Array<Strings>} ids ids of agents
+ * @param {String} apiId API id
+ */
+export async function buildAgentsTable(context, printer: ReportPrinter, agentIDs: string[], apiId: string, multi = false) {
+  const dateFormat = await context.core.uiSettings.client.get('dateFormat');
+  if (!agentIDs || !agentIDs.length) return;
+  log('reporting:buildAgentsTable', `${agentIDs.length} agents for API ${apiId}`, 'info');
+  try {
+    let agentRows = [];
+    if (multi) {
+      try {
+        const agentsResponse = await context.wazuh.api.client.asCurrentUser.request(
+          'GET',
+          `/groups/${multi}/agents`,
+          {},
+          { apiHostID: apiId }
+        );
+        const agentsData =
+          agentsResponse &&
+          agentsResponse.data &&
+          agentsResponse.data.data &&
+          agentsResponse.data.data.affected_items;
+        agentRows = (agentsData || []).map((agent) => ({
+          ...agent,
+          manager: agent.manager || agent.manager_host,
+          os:
+            agent.os && agent.os.name && agent.os.version
+              ? `${agent.os.name} ${agent.os.version}`
+              : '',
+        }));
+      } catch (error) {
+        log(
+          'reporting:buildAgentsTable',
+          `Skip agent due to: ${error.message || error}`,
+          'debug'
+        );
+      }
+    } else {
+      for (const agentID of agentIDs) {
+        try {
+          const agentResponse = await context.wazuh.api.client.asCurrentUser.request(
+            'GET',
+            `/agents`,
+            { params: { q: `id=${agentID}` } },
+            { apiHostID: apiId }
+          );
+          const [agent] = agentResponse.data.data.affected_items;
+          agentRows.push({
+            ...agent,
+            manager: agent.manager || agent.manager_host,
+            os: (agent.os && agent.os.name && agent.os.version) ? `${agent.os.name} ${agent.os.version}` : '',
+            lastKeepAlive: moment(agent.lastKeepAlive).format(dateFormat),
+            dateAdd: moment(agent.dateAdd).format(dateFormat)
+          });
+        } catch (error) {
+          log(
+            'reporting:buildAgentsTable',
+            `Skip agent due to: ${error.message || error}`,
+            'debug'
+          );
+        }
+      }
+    }
+    printer.addSimpleTable({
+      columns: [
+        { id: 'id', label: 'ID' },
+        { id: 'name', label: 'Name' },
+        { id: 'ip', label: 'IP' },
+        { id: 'version', label: 'Version' },
+        { id: 'manager', label: 'Manager' },
+        { id: 'os', label: 'OS' },
+        { id: 'dateAdd', label: 'Registration date' },
+        { id: 'lastKeepAlive', label: 'Last keep alive' },
+      ],
+      items: agentRows,
+    });
+  } catch (error) {
+    log('reporting:buildAgentsTable', error.message || error);
+    return Promise.reject(error);
+  }
+}
```

Comment:
**issue**: This method was affected in a previous PR: https://github.com/wazuh/wazuh-kibana-app/pull/4350 changing some logic due to bugs. We should update it.